cli/src/pcluster/models/imagebuilder.py

```diff
@@ -213,29 +234,28 @@ def _get_custom_bucket(self):
 
     def _get_artifact_dir(self):
         """Get artifact directory from image tag or stack tag."""
+        s3_artifact_dir = None
         try:
-            image_info = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
-            self.__s3_artifact_dir = search_tag(image_info, PCLUSTER_S3_IMAGE_DIR_TAG)
-            if self.__s3_artifact_dir is None:
-                raise ImageNotFoundError
-        except AWSClientError as e:
-            if not isinstance(e, ImageNotFoundError):
+            s3_artifact_dir = self.image.s3_artifact_directory
+        except ImageBuilderActionError as e:
+            if f"Image {self.image_name} does not exist." not in str(e):
                 LOGGER.error("Unable to find tag %s in image %s.", PCLUSTER_S3_IMAGE_DIR_TAG, self.image_name)
                 raise ImageBuilderActionError(f"Unable to get artifact directory from image {self.image_name} tag. {e}")
+        if s3_artifact_dir is None:
+            try:
+                s3_artifact_dir = self.stack.s3_artifact_directory
 
-        try:
-            self.__s3_artifact_dir = self.stack.s3_artifact_directory
-
-            if self.__s3_artifact_dir is None:
-                raise AWSClientError(
-                    function_name="_get_artifact_dir",
-                    message="No artifact directory found in image tag and cloudformation stack tag.",
+                if s3_artifact_dir is None:
+                    raise ImageBuilderActionError(
+                        "No artifact directory found in image tag and cloudformation stack tag."
+                    )
+            except ImageBuilderActionError as e:
+                raise ImageBuilderActionError(
```

Comment:
As you can see there is something strange here, you are raising ImageBuilderActionError in the 249 line then you're caching it here.
If you have NonExistingStackError you can distinguish the two error cases in a simpler way.

Reply:
done